scripts/backup_with_rotation_original.sh

```
@@ -0,0 +1,47 @@
+if [ $# -eq 0 ] || [ ! -d "$1" ]; then
+    echo "Error: Please provide a valid directory path as a command-line argument."
+    display_usage
+    exit 1
+fi
+
+# Assign source directory from command-line argument
+source_dir="$1"
+
+# Function to create a timestamped backup and zip it
+function create_backup {
+    local timestamp=$(date '+%Y-%m-%d_%H-%M-%S')
```

Shellcheck

[warning] Declare and assign separately to avoid masking return values.

(SC2155)